File: app/src/main/java/com/example/helper/todo/TodoActivity.java
package com.example.helper.todo;

import android.content.SharedPreferences;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.util.Log;

import com.example.helper.R;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.Set;

import static com.example.helper.MainActivity.PREF_NAME;

public class TodoActivity extends AppCompatActivity {

    RecyclerView recyclerView;
    RecyclerView.LayoutManager layoutManager;
    TodoAdapter todoAdapter;
    ArrayList todos;
    SharedPreferences prefs;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_to_do);

        initialiseTodos();
        initialiseRecyclerView();
    }

    private void initialiseTodos() {
        todos = new ArrayList();

        todos = getIntent().getParcelableArrayListExtra("todos");

        prefs = getSharedPreferences(PREF_NAME, MODE_PRIVATE);
        Set<String> todosSet = prefs.getStringSet("todos", Collections.singleton(""));
        Log.d("Todos", "" + todosSet);
//        todos = (ArrayList) todosSet;
    }

    private void initialiseRecyclerView() {
        recyclerView = findViewById(R.id.todo_recycler_view);
        layoutManager = new LinearLayoutManager(getApplicationContext(), RecyclerView.VERTICAL, false);
        todoAdapter = new TodoAdapter(getApplicationContext(), (List<TodoModel>) todos);
        recyclerView.setLayoutManager(layoutManager);
        recyclerView.setAdapter(todoAdapter);

    }

}

File: app/src/main/java/com/example/helper/todo/TodoActivity.kt
package com.example.helper.todo

import android.content.Context
import android.content.SharedPreferences
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.util.Log

import com.example.helper.R


import com.example.helper.MainActivity.PREF_NAME

class TodoActivity : AppCompatActivity() {

    lateinit var recyclerView: RecyclerView
    lateinit var layoutManager: RecyclerView.LayoutManager
    lateinit var todoAdapter: TodoAdapter
    private lateinit var todos: List<String>
    private lateinit var prefs: SharedPreferences

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_to_do)

        initialiseTodos()
        initialiseRecyclerView()
    }

    private fun initialiseTodos() {

        todos = intent.getParcelableArrayListExtra("todos")

        prefs = getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE)
        val todosSet = prefs.getStringSet("todos", setOf(""))
        Log.d("Todos", "" + todosSet!!)
        //        todos = (ArrayList) todosSet;
    }

    private fun initialiseRecyclerView() {
        recyclerView = findViewById(R.id.todo_recycler_view)
        layoutManager = LinearLayoutManager(applicationContext, RecyclerView.VERTICAL, false)
        todoAdapter = TodoAdapter(applicationContext, todos as List<TodoModel>)
        recyclerView.layoutManager = layoutManager
        recyclerView.adapter = todoAdapter

    }

}
